src/main/webapp/go.js

```diff
@@ -150,30 +185,35 @@ function toggleHintButton(hide) { //eslint-disable-line
  * Disable lint check because getHint() is called from go.html.
  */
 function getHint() { //eslint-disable-line
-  if (destIndex >= 0) {
-    const hintArr = huntArr[destIndex].hints;
-    if (hintIndex < hintArr.length) {
-      updateMessage(HINT_DISPLAY, hintArr[hintIndex]);
-      hintIndex++;
-    }
+  const nextHint = hunt.getNextHint();
+  if (nextHint != '') {
+    updateMessage(HINT_DISPLAY, nextHint);
   }
 }
 
 /**
  * Determines whether the user entered the correct destination, and
  * adjusts the display accordingly.
+ * Disable lint check because handleDestinationAnswer() is called
+ * from go.html.
  */
-function handleDestinationAnswer() {
-  fetch(GUESS_URL).then((response) => response.json()).then((guess) => {
-    if (guess === huntArr[destIndex].name) {
-      toggleProceedButton(/* hide = */ false);
-      toggleHintButton(/* hide = */ true);
+function handleDestinationAnswer() { //eslint-disable-line
+  if (hunt.hasNotStarted()) { // User has not yet started the hunt.
+    return;
+  }
+  const userGuess = document.getElementById(GUESS_DISPLAY).value;
+  const queryStr = GUESS_URL + '?user-input=' + userGuess +
+      '&answer=' + hunt.getName();
+  fetch(queryStr).then((response) => response.json()).then((correctGuess) => {
+    if (correctGuess) {
+      toggleProceedButton(false);
+      toggleHintButton(true);
       updateMessage(SUBMIT_DISPLAY, CORRECT_MSSG);
-      updateMessage(RIDDLE_DISPLAY, huntArr[destIndex].name + ': ' +
-          huntArr[destIndex].description);
-      addMarkerToMap(huntArr[destIndex].lat, huntArr[destIndex].lng,
-          huntArr[destIndex].name);
-    } else if (guess.length != 0) {
+      updateMessage(RIDDLE_DISPLAY, hunt.getName() + ': ' +
```

Comment:
these all pertain to the current destination, so maybe a better name is getCurrentDestinationName(), etc.

OR have getCurrentDestination() return a Destination class, which then has getName() methods on it